### ISSUE TITLE
Count(<All>) query is discarding filters on Mongo plugin

### ISSUE THREAD

**Issue by raj-s-gouda** [apache/drill#2562]
**Describe the bug**
When below query is run on MongoDB plugin using Drill 1.19  it is returning an invalid count as output
 select count(*) from Table_A where Col_1 <= 3

**To Reproduce**
Data: Table_A
Col_1  |  Col_2
-----------------------
1         |   A
2         |   B
3         |   C
4         |   D
5         |   E

**Expected behavior**
Count  : 3

Actual Result : 5

**Comment by luocooong** [comment]
@raj-s-gouda Could you please post the mongo data (json)? Please mask any privacy information.

**Comment by cgivre** [comment]
Could you also please try on Drill 1.20?  There's been a lot of work done on both the JSON reader and the Mongo plugin since Drill 1.19.

**Comment by raj-s-gouda** [comment]
> 

[
	{
		
		"Col_1" : 1,
		"Col_2" : A
	},
	{
		
		"Col_1" : 2,
		"Col_2" : B
	},
	{
		
		"Col_1" : 3,
		"Col_2" : C
	},
	{
		
		"Col_1" : 4,
		"Col_2" : D
	},
	{
		
		"Col_1" : 5,
		"Col_2" : E
	}

]

**Comment by raj-s-gouda** [comment]
@cgivre : Yes. It is working fine with 1.20. But I am facing other compatibility issues with that. Hence looking for the solution in 1.19

**Comment by jnturton** [comment]
@raj-s-gouda we only have the bandwidth to maintain one stable release and that's now 1.20.x. So you can file Jiras for the other issues blocking your upgrade to 1.20 (especially if they are regressions) or you can try to backport Mongo-related fixes to 1.19 yourself and do your own build...

**Comment by cgivre** [comment]
Fixed in Drill 1.20.

**Comment by raj-s-gouda** [comment]
@jnturton No, not this one. It is https://github.com/apache/drill/issues/2563

Backporting https://github.com/apache/drill/commit/98378a992034b9d101fa010547fe6c5a78dbf7db to 1.19 has solved my problem
